Add Toast component with customizable features and Sonner integration.

iamshubhransh opened the pull request.

iamshubhransh committed:
Add Toast component with customizable features and Sonner integration.

shyamkumawat088 commented on package.json:
```diff
@@ -161,7 +161,9 @@
-        "yup": "^1.6.1"
+        "yup": "^1.6.1",
+        "sonner": "2.0.7",
+        "next-themes": "0.4.6"
```
why we need this package?

iamshubhransh committed:
Refactor Toast component to use 'variant' and 'color' props instead of 'emphasis' and 'type'. Update related constants, styles, and utility functions for improved clarity and consistency. Enhance documentation and tests to reflect these changes.

iamshubhransh committed:
Merge branch 'master' into add-toast-component

shyamkumawat088 commented on stories/elements/Toast/index.stories.tsx:
```diff
@@ -0,0 +1,184 @@
+    leftIcon: 'bell',
+    color: 'brand',
+    title: 'Notification',
+};
```
add new line

shyamkumawat088 commented on ui/elements/Toast/constants.ts:
```diff
@@ -0,0 +1,185 @@
+export enum TOAST_MOBILE_PLACEMENT_ENUM {
+    TOP = 'top',
+    BOTTOM = 'bottom',
+}
```
create single enum for all position and consider all the below positions
```   
TOP_LEFT = 'top-left',
    TOP_RIGHT = 'top-right',
    BOTTOM_LEFT = 'bottom-left',
    BOTTOM_RIGHT = 'bottom-right',
   TOP_CENTER = 'top-center',
    BOTTOM_CENTER = 'bottom-center',
```

shyamkumawat088 commented on ui/elements/Toast/constants.ts:
```diff
@@ -0,0 +1,185 @@
+    MOBILE = 'mobile',
+}
+
+export enum TOAST_VARIANTS_ENUM {
+    LOW = 'low',
+    MEDIUM = 'medium',
+    HIGH = 'high',
+}
+
```
use variants as 
primary, secondary & tertiary.

shyamkumawat088 commented on ui/elements/Toast/constants.ts:
```diff
@@ -0,0 +1,185 @@
+export enum TOAST_DEVICE_ENUM {
+    DESKTOP = 'desktop',
+    MOBILE = 'mobile',
+}
```
it should not be on the basis of device.

iamshubhransh committed:
Enhance Toast component with rich variant support and improved placement options

- Introduced a new 'rich' layout type for the Toast component, allowing for a card-style design with title, subtitle, and custom images.
- Updated the Toast component to handle 'showCTA' and 'showSecondaryCTA' properties dynamically based on the layout type.
- Added new placement options for desktop and mobile, including 'top-center' and 'bottom-center'.
- Refactored styles to accommodate the new rich variant, ensuring proper dimensions and alignment.
- Enhanced documentation and tests to reflect the new features and ensure functionality across different devices and placements.

iamshubhransh committed:
Refactor Toast component for improved mobile responsiveness and styling

- Updated the Toast component to ensure the rich variant is fluid on mobile, filling its container with margins controlled by the xOffset prop.
- Adjusted styles for mobile to allow subtitles to wrap up to two lines and truncate with an ellipsis, enhancing readability.
- Modified the actions row to allow shrinking, preventing overflow of secondary CTAs beyond the toast's edge.
- Enhanced tests to verify mobile behavior for rich variants, ensuring correct width calculations and subtitle handling.

iamshubhransh committed:
Remove next-themes dependency and update Toast component for improved variant handling

- Removed the next-themes package from package.json and package-lock.json.
- Refactored Toast component to utilize new variant naming conventions: 'primary', 'secondary', and 'tertiary'.
- Updated Toast stories to reflect changes in variant levels and improved documentation for clarity.
- Enhanced Toast styles for better responsiveness and alignment across different devices.

shyamkumawat088 commented on ui/core/icons/iconTags.ts:
```diff
@@ -49,14 +49,8 @@ export const ICON_TAGS = {
-    'currency-filled': ['Finance'],
     'currency-rupee-filled': ['Finance'],
     'currency-rupee': ['Finance'],
-    currency: ['Finance'],
-    'eye-off-filled': ['Media'],
-    'eye-off': ['Media'],
-    'eye-on-filled': ['Media'],
-    'eye-on': ['Media'],
```
please check this changes, why it is removed. seems un-related with PR.

shyamkumawat088 commented on ui/core/icons/iconTypes.ts:
```diff
@@ -49,14 +49,8 @@ export type grauityIconName =
-    | 'currency-filled'
     | 'currency-rupee-filled'
     | 'currency-rupee'
-    | 'currency'
-    | 'eye-off-filled'
-    | 'eye-off'
-    | 'eye-on-filled'
-    | 'eye-on'
```
changes seems un-related.

shyamkumawat088 commented on stories/elements/Toast/Gallery.stories.tsx:
```diff
@@ -0,0 +1,253 @@
+const toastColors = ['warning', 'brand', 'neutral', 'success', 'error'] as const;
+const variantLevels = ['primary', 'secondary', 'tertiary'] as const;
+
+export const AllVariants = () => (
+    <div style={{ display: 'flex', flexDirection: 'column', gap: '24px', padding: '20px' }}>
+        <h2>Toast Gallery - All Variants</h2>
+
+        {variantLevels.map(variant => (
```
please show only gallery, doc, component only

shyamkumawat088 commented on stories/elements/Toast/Gallery.stories.tsx:
```diff
@@ -0,0 +1,253 @@
+        <h2>Toast Gallery - All Variants</h2>
+
+        {variantLevels.map(variant => (
+            <div key={variant} style={{ display: 'flex', flexDirection: 'column', gap: '12px' }}>
+                <h3 style={{
+                    textTransform: 'capitalize',
+                    margin: '8px 0 4px 0',
+                    fontSize: '14px',
+                    fontWeight: 500,
+                    color: '#666',
+                }}>
```
i dont think this styles are required, please check.

shyamkumawat088 commented on stories/elements/Toast/Gallery.stories.tsx:
```diff
@@ -0,0 +1,253 @@
+
+const richReferralImage = (
+    <div
+        aria-hidden="true"
+        style={{
+            display: 'flex',
+            alignItems: 'center',
+            justifyContent: 'center',
+            width: '44px',
+            height: '44px',
+            borderRadius: '50%',
+            background: 'linear-gradient(135deg, #ef9d1a 0%, #d0850b 100%)',
+            color: '#ffffff',
+            fontFamily: '\'Mona Sans\', sans-serif',
+            fontWeight: 800,
+            fontSize: '11px',
```
check for all other occurrences in this file

shyamkumawat088 commented on ui/elements/Toast/index.ts:
```diff
@@ -0,0 +1,26 @@
+    ToastType,
+    ToastVariant,
+} from './types';
+export {
+    getCloseButtonProps,
+    getCTAButtonProps,
+    getPlacementStyles,
+    getToastColors,
+    getToastIcon,
+} from './utils';
```
this functions are not required to export. as this function are used internally to get the configuration for variants and colors.

shyamkumawat088 commented on ui/elements/Toast/Toast.tsx:
```diff
@@ -0,0 +1,348 @@
+        variant = TOAST_VARIANTS_ENUM.SECONDARY,
+        color = 'neutral',
+        showLeftIcon = true,
+        leftIcon,
```
rename it to something like primaryIcon.

shyamkumawat088 commented on ui/elements/Toast/Toast.tsx:
```diff
@@ -0,0 +1,348 @@
+        type = TOAST_TYPES_ENUM.SIMPLE,
+        variant = TOAST_VARIANTS_ENUM.SECONDARY,
+        color = 'neutral',
+        showLeftIcon = true,
```
as we are already passing leftIcon prop so this prop `showLeftIcon` is not required.

shyamkumawat088 commented on ui/elements/Toast/Toast.tsx:
```diff
@@ -0,0 +1,348 @@
+        ctaText: ctaTextProp,
+        onCTAClick,
+        onClose,
+        title: titleProp,
+        subtitle: subtitleProp,
```
make sure this will take just string

shyamkumawat088 commented on ui/elements/Toast/Toast.tsx:
```diff
@@ -0,0 +1,348 @@
+        showLeftIcon = true,
+        leftIcon,
+        showCloseIcon = true,
+        showCTA: showCTAProp,
+        ctaText: ctaTextProp,
```
use ctaText only instead of taking showCTA as well, also rename it to something relevant
